src/cloudflare/internal/images-api.ts

```diff
@@ -303,6 +327,165 @@ function chainStreams<T>(streams: ReadableStream<T>[]): ReadableStream<T> {
   return outputStream;
 }
 
+function concatUint8Arrays(a: Uint8Array, b: Uint8Array): Uint8Array {
+  const result = new Uint8Array(a.length + b.length);
+  result.set(a, 0);
+  result.set(b, a.length);
+  return result;
+}
+
+class Base64Error extends Error {
+  public constructor(cause: unknown) {
+    if (cause instanceof Error) {
+      super(`base64 error: ${cause.message}`, { cause });
+    } else {
+      super('unknown base64 error');
+    }
+  }
+}
+
+function createBase64EncoderTransformStream(
+  maxEncodeChunkSize: number = 32 * 1024 + 1
+): TransformStream<Uint8Array, Uint8Array> {
+  if (maxEncodeChunkSize % 3 != 0) {
+    // Ensures that chunks don't require padding
+    throw new Error('maxChunkSize must be a multiple of 3');
+  }
+
+  let buffer: Uint8Array | null = null;
+  const asciiEncoder = new TextEncoder();
+
+  const toBase64: (buf: Uint8Array) => Uint8Array = (buf) => {
+    const binaryString = String.fromCharCode.apply(null, Array.from(buf));
+
+    const base64String = btoa(binaryString);
+    const base64Bytes = asciiEncoder.encode(base64String);
```

Comment:
Hmm... we should probably look at providing a `cloudflare-internal:...` API to do the base64 encoding more efficiently. If nodejs_compat is on then I we could just use `Buffer`s built-in base64 support. We could expose the same thing via a `cloudflare-internal` for times when `nodejs_compat` is not enabled.